examples/gno.land/p/demo/bitmap/bitmap.gno

```diff
@@ -0,0 +1,40 @@
+package bitmap
+
+// A simple implementation of Bitmap
+
+// Bitmap represents a bitmap using a slice of bytes
+type Bitmap struct {
+	data []byte
+}
+
+// NewBitmap creates a new Bitmap with a specific size (in bits)
+func New(size uint64) *Bitmap {
+	byteSize := (size + 7) / 8 // Calculate the number of bytes needed
+	return &Bitmap{
+		data: make([]byte, byteSize),
+	}
+}
+
+// Set sets the bit at the given index (0-based)
+func (b *Bitmap) Set(index uint64) {
```

Comment:
```suggestion
func (b *Bitmap) Set(index uint64) error {
```

You can create an additional `MustSet()` function that panics if needed.

Reply:
good idea, it will do it right now

Reply:
@moul i have worked on refactoring, please take a look